src/Chainweb/Pact/PactService/Checkpointer.hs

```diff
@@ -654,3 +640,18 @@ heightProgress initialHeight ref logFun = forever $ do
       <> ", current height: " <> sshow h
     threadDelay (20 * 1_000000)
 
+failOnTooLowRequestedHeight
+    :: forall tbl
+    . HasCallStack
+    => CanReadablePayloadCas tbl
+    => BlockHeader
+    -> Maybe Limit
+    -> BlockHeader
+    -> PactServiceM tbl ()
+failOnTooLowRequestedHeight parent (Just limit) lastHeader
+    | parentHeight + 1 + (BlockHeight $ _limit limit) < lastHeight = -- need to stick with addition because Word64
+        throwM $ RewindLimitExceeded limit parentHeight lastHeight parent
+  where
+    parentHeight = _blockHeight parent
+    lastHeight = _blockHeight lastHeader
```

Comment:
Less logic in the guards is :ok_hand: 
```suggestion
failOnTooLowRequestedHeight parent (Just limit) lastHeader
    | parentHeight + 1 + limitHeight < lastHeight = -- need to stick with addition because Word64
        throwM $ RewindLimitExceeded limit parentHeight lastHeight parent
  where
    limitHeight = BlockHeight $ _limit Limit
    parentHeight = _blockHeight parent
    lastHeight = _blockHeight lastHeader
```

Reply:
Thanks! 👍